pymongo/server.py

```diff
@@ -89,7 +89,7 @@ def close(self) -> None:
                 )
             )
         self._monitor.close()
-        self._pool.reset_without_pause()
+        self._pool.close()
```

Comment:
why do we opt to close here rather than the old behavior?

Reply:
See https://jira.mongodb.org/browse/PYTHON-4166.